pkg/kubeapiserver/options/authentication.go

```diff
@@ -245,6 +246,9 @@ func (s *BuiltInAuthenticationOptions) AddFlags(fs *pflag.FlagSet) {
 		fs.StringVar(&s.ServiceAccounts.Issuer, "service-account-issuer", s.ServiceAccounts.Issuer, ""+
 			"Identifier of the service account token issuer. The issuer will assert this identifier "+
 			"in \"iss\" claim of issued tokens. This value is a string or URI.")
+
+		fs.StringSliceVar(&s.ServiceAccounts.APIAudiences, "service-account-api-audiences", s.ServiceAccounts.APIAudiences, ""+
```

Comment:
Sorry if this was discussed previously, but why multiple audiences?

Reply:
This will be nice when:
1) API servers are accessed from multiple URI's (e.g. behind a loadbalancer and directly)
1) Someone wants to change the name of the apiserver (e.g. changing DNS name of LB) which is a special case of 1)
1) If someone wants to have a universal and specific identifier for their cluster API (e.g. https://kubernetes.default.svc and https://cluster-1.dev.example.com)

Reply:
> API servers are accessed from multiple URI's (e.g. behind a loadbalancer and directly)

All for that. This is one fo the limitations of the OpenID Connect spec that's caused us pain.

The JWT validation requires all the audiences to be present in the JWT: https://github.com/square/go-jose/blob/v2.1.3/jwt/validation.go#L73

It sounds like we only want to require one audience out of the set.

Reply:
Fixed.